Add `noplayground` annotation to guide snippets

Hofer-Julian opened the pull request:
None of them can be run in the playground

davidhewitt reviewed (commented):
Thanks for this!

Two quick thoughts:
- From https://rust-lang.github.io/mdBook/format/mdbook.html#rust-playground it looks like we may be able to do this globally in the `book.toml` configuration, I wonder if that's preferable?
- No need for a CHANGELOG entry for a guide-only change; I've added the `CI-skip-changelog` label, please remove the newsfragment.

Hofer-Julian commented:
> From https://rust-lang.github.io/mdBook/format/mdbook.html#rust-playground it looks like we may be able to do this globally in the book.toml configuration, I wonder if that's preferable?

Yes, I think so. Added it and it works as expected.

> No need for a CHANGELOG entry for a guide-only change; I've added the CI-skip-changelog label, please remove the newsfragment.

Done

davidhewitt reviewed (approved):
Thanks 👍
